lib/formatters/tapFormatter.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+/**
+ * @type {import('stylelint').Formatter}
+ */
+const tapFormatter = (results) => {
+	let lines = [`TAP version 13\n1..${results.length}`];
+
+	results.forEach((result, index) => {
+		lines.push(
+			`${result.errored ? 'not ok' : 'ok'} ${index + 1} - ${result.ignored ? 'ignored ' : ''}${
+				result.source
+			}`,
+		);
+
+		if (result.warnings.length > 0) {
+			lines.push('---', 'messages:');
+
+			result.warnings.forEach((warning) => {
+				lines.push(
+					` - message: "${warning.text}"`,
+					`   severity: ${warning.severity}`,
+					`   data:`,
+					`     line: ${warning.line}`,
+					`     column: ${warning.column}`,
+					`     ruleId: ${warning.rule}`,
+				);
+			});
+
+			lines.push('---');
```

Comment:
@ybiquitous

According to https://testanything.org/tap-version-14-specification.html#yaml-diagnostics it should be `...`, right?

Reply:
We have implemented the version 13, not version 14:

https://github.com/stylelint/stylelint/blob/eb6d61357765f39ab7dceeb71d7749e55422564e/lib/formatters/tapFormatter.mjs#L7

But it seems v13 also requires an ending `...`, according to <https://testanything.org/tap-version-13-specification.html>:

> If the test line is immediately followed by an indented block beginning with `/^\s+---/` and ending with `/^\s+.../` that block will be interpreted as an inline YAML document.

So, it makes sense for me to fix `---` to `...`. Additionally, we might be able to update the TAP version from 13 to 14.

Can you create an issue or pull request if you have time?

Reply:
I was reviewing which formatter would have the `fixed` mention.
see https://github.com/stylelint/stylelint/issues/4664#issuecomment-2159060803

> Can you create an issue or pull request if you have time?

Not right away no, but what I wonder is why it wasn't picked up for all these years.
The tap consumers must be very lax.
This is weird.

Reply:
Actually, I am planning to send a PR this weekend.